src/solidity-frontend/solidity_convert.cpp

```diff
@@ -5182,6 +5233,54 @@ bool solidity_convertert::get_expr(
       new_expr = call;
       break;
     }
+    // * special: new D{value: amount}(4), handle the {value: amount} part
+    if (
+      callee_expr_json["nodeType"] == "FunctionCallOptions" &&
+      callee_expr_json["expression"]["nodeType"] == "NewExpression")
+    {
+      exprt new_obj;
+      // 1. construct new object
+      if (get_new_object_ctor_call(
+            callee_expr_json["expression"], true, new_obj))
+        return true;
+      // 2. if it has value option, we need to model the transaction
+      if (
+        callee_expr_json["options"].contains("name") &&
+        callee_expr_json["options"]["name"] == "value")
+      {
+        for (const auto &opt : callee_expr_json["options"])
+        {
+          locationt loc;
+          get_location_from_node(expr, loc);
+
+          exprt this_obj;
+          if (get_ctor_decl_this_ref(expr, this_obj))
```

Comment:
Why ctor only? Can't this "new expression with options" happen in a normal function body rather than a constructor?